blockchain/projects.go

```diff
@@ -55,20 +55,34 @@ type Projects struct {
 	accountsNumber int
 }
 
-// Reset the blockchain state.
-func (p *Projects) Reset(project *model.Project) ([]*model.Account, error) {
-	p.emulatorCache.reset(project.ID)
+// Reset the blockchain state and return the new account models
+func (p *Projects) Reset(projectID uuid.UUID, em *blockchain) ([]*model.Account, error) {
+	var project model.Project
+	err := p.store.GetProject(projectID, &project)
+	if err != nil {
+		return nil, err
+	}
+
+	p.emulatorCache.reset(projectID)
 
-	err := p.store.ResetProjectState(project)
+	err = p.store.ResetProjectState(&project)
 	if err != nil {
 		return nil, err
 	}
 
-	accounts, err := p.CreateInitialAccounts(project.ID)
+	accounts, err := p.createInitialAccounts(projectID)
 	if err != nil {
 		return nil, err
 	}
 
+	// Reload emulator
```

Comment:
Why is this needed?

Reply:
I think it is required as Reset creates new initial accounts on a new emulator to reset the state and the current emulator being used would be stale if not reloaded (for example in deploy contract). It probably won't be needed when we change contract deployment logic but I'll create another ticket for that.